src/main/java/com/pub/wysio/controller/CustomerController.java

```diff
@@ -0,0 +1,49 @@
+package com.pub.wysio.controller;
+
+import com.pub.wysio.entity.Customer;
+import com.pub.wysio.service.CustomerService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+public class CustomerController {
+    @Autowired
+    private CustomerService service;
+
+    @PostMapping("/addCustomer")
```

Comment:
You added the `customer` string to every endpoint. This is good thinking, because we want to differentiate between endpoints (e.g. Customer and Fight endpoints). But instead of this approach, you can use the [request mapping](https://www.baeldung.com/spring-controller-vs-restcontroller#spring-mvc-rest-controller) annotation on the controller, so the controller in itself would add the customer string to every endpoint

meaning instead of `POST localhost:8080/addCustomer` endpoint you would have:

`GET localhost:8080/customer/{id}` to get by id
`POST localhost:8080/customer` to create
`PUT localhost:8080/customer/{id}`  to update

Reply:
note that I modified the endpoint names. This is a common practice, because you could have multiple endpoints with the same name, but with different HTTP methods. This way there is no more string literals than necessary. I suggest you to follow this approach as well

Reply:
Almost good. Now you have `/customer` defined in every endpoint, which is a bit redundant. You could do it like this:

```java
@RequestMapping("customer")

// ... rest of the code

    @GetMapping
    public List<Customer> findAllCustomers() { 
        //...
    }

    @GetMapping("{id}")
    public Customer findCustomerById(@PathVariable Integer id) {
        //...
    }
```

This results in the same endpoint names, but you only define the `customer` string literal once: on the controller level. This way every endpoint will have the customer string by default, you cannot create and endpoint that does not have it. Also you by only defining the string literal once, there is less space for a typo, when creating a new endpoint.